client/my-sites/controller.js

```diff
@@ -305,81 +323,61 @@ module.exports = {
 		} else {
 			// if sites has fresh data and siteID is invalid
 			// redirect to allSitesPath
-			if ( sites.fetched || ! sites.fetching ) {
+			if ( ! isRequestingSite( getState(), siteID ) ) {
 				return page.redirect( allSitesPath );
 			}
 
 			let waitingNotice;
 			const selectOnSitesChange = () => {
 				// if sites have loaded, but siteID is invalid, redirect to allSitesPath
-				if ( sites.select( siteID ) ) {
-					sites.initialized = true;
+				dispatch( setSelectedSiteId( siteID ) );
+				if ( getSite( getState(), siteID ) ) {
 					onSelectedSiteAvailable( context );
 					if ( waitingNotice ) {
 						notices.removeNotice( waitingNotice );
 					}
-				} else if ( ( currentUser.visible_site_count !== sites.getVisible().length ) ) {
-					sites.initialized = false;
+				} else if ( ( currentUser.visible_site_count !== getVisibleSites( getState() ).length ) ) {
 					waitingNotice = notices.info( i18n.translate( 'Finishing set up…' ), { showDismiss: false } );
-					sites.once( 'change', selectOnSitesChange );
-					sites.fetch();
+					dispatch( {
+						type: SITES_ONCE_CHANGED,
```

Comment:
Is the intention here to dispatch this event, the first time sites is updated per page load?